artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/LivePageCacheImpl.java

```diff
@@ -28,15 +28,49 @@
 /**
  * This is the same as PageCache, however this is for the page that's being currently written.
  */
```

Comment:
Isn't this mixing the collection implementation itself into the LivePageCache?

isn't there a way to implement this logic into its own structure? Like PageCache using a generic ChunkArray (a name I just came up here)?

I'm a bit concerned on maintaining the business side of this issue (that is the PageCache) with the speedy implementation of a collection.

Reply:
@franz1981 Id agree with @clebertsuconic here, a bit like what ive done with priority consumers, i ended up splitting out the collections logic, which has ended up making it cleaner, and easier to reason with. (and as youve marked out on its pr, more testable ;) )

Reply:
Makes sense :+1: